command/kv_delete.go

```diff
@@ -134,29 +184,14 @@ func (c *KVDeleteCommand) Run(args []string) int {
 }
 
 func (c *KVDeleteCommand) deleteV2(path, mountPath string, client *api.Client) (*api.Secret, error) {
-	var err error
-	var secret *api.Secret
-	switch {
-	case len(c.flagVersions) > 0:
+	if len(c.flagVersions) > 0 {
 		path = addPrefixToKVPath(path, mountPath, "delete")
-		if err != nil {
-			return nil, err
-		}
-
 		data := map[string]interface{}{
 			"versions": kvParseVersionsFlags(c.flagVersions),
 		}
 
-		secret, err = client.Logical().Write(path, data)
-	default:
-
-		path = addPrefixToKVPath(path, mountPath, "data")
-		if err != nil {
-			return nil, err
-		}
-
-		secret, err = client.Logical().Delete(path)
+		return client.Logical().Write(path, data)
```

Comment:
Style: this is an early non-error return and it's often recommend to avoid these (since they could be missed by the next developer and could cause unintentional bugs). 

While the previous style with the switch statement was more verbose, I found it a bit easier to read (both `client.Logical().Write()` and `client.Logical().Delete()` were on the same level. The only thing that was truly redundant is the `if err != nil {..}` check.

Reply:
That makes sense, okay I've restored it but without the err checks.